pkg/sentry/fsimpl/fuse/init.go

```diff
@@ -0,0 +1,294 @@
+			linux.FUSE_DO_READDIRPLUS |
+			linux.FUSE_READDIRPLUS_AUTO |
+			linux.FUSE_ASYNC_DIO |
+			linux.FUSE_WRITEBACK_CACHE |
+			linux.FUSE_NO_OPEN_SUPPORT |
+			linux.FUSE_PARALLEL_DIROPS |
+			linux.FUSE_HANDLE_KILLPRIV |
+			linux.FUSE_POSIX_ACL |
+			linux.FUSE_ABORT_ERROR |
+			linux.FUSE_MAX_PAGES |
+			linux.FUSE_CACHE_SYMLINKS |
+			linux.FUSE_NO_OPENDIR_SUPPORT |
+			linux.FUSE_EXPLICIT_INVAL_DATA,
+	}
+
+	return fs.fuseConn.NewRequest(creds, pid, 0, linux.FUSE_INIT, &in)
+}
+
+// Process the FUSE_INIT reply from the FUSE server.
+// The handling process follows the rules of the FUSE protocol implementation of unix kernel.
+func (fs *filesystem) initProcessReply(k *kernel.Kernel, creds *auth.Credentials, out *linux.FUSEInitOut) error {
+	// No support for old major fuse versions.
+	// This behavior is consistent with the Linux kernel (v5.8).
+	if out.Major != linux.FUSE_KERNEL_VERSION {
+		fs.fuseConn.ConnError = true
+	} else {
+		// No support for limits before minor version 13.
+		if out.Minor >= 13 {
+			fs.initProcessLimits(k, creds, out)
+		}
+
+		// No support for the following flags before minor version 6.
+		if out.Minor >= 6 {
+			if out.Flags&linux.FUSE_ASYNC_READ != 0 {
+				fs.fuseConn.AsyncRead = true
+			}
+
+			if out.Flags&linux.FUSE_POSIX_LOCKS == 0 {
+				fs.fuseConn.NoLock = true
+			}
+
+			// No support for FLOCK flag before minor version 17.
+			if out.Minor >= 17 {
+				if out.Flags & linux.FUSE_FLOCK_LOCKS == 0 {
+					fs.fuseConn.NoFLock = true
+				}
+			} else {
+				if out.Flags & linux.FUSE_POSIX_LOCKS == 0 {
+					fs.fuseConn.NoFLock = true
+				}
+			}
+
+			if out.Flags & linux.FUSE_ATOMIC_O_TRUNC != 0 {
+				fs.fuseConn.AtomicOTrunc = true
+			}
+
+			// No support for EXPORT flag before minor version 9.
+			if out.Minor >= 9 {
+				if out.Flags & linux.FUSE_EXPORT_SUPPORT != 0 {
+					fs.fuseConn.ExportSupport = true
+				}
+			}
+
+			if out.Flags & linux.FUSE_BIG_WRITES != 0 {
+				fs.fuseConn.BigWrites = true
+			}
+
+			if out.Flags & linux.FUSE_DONT_MASK != 0 {
+				fs.fuseConn.DontMask = true
+			}
+
+			if out.Flags & linux.FUSE_AUTO_INVAL_DATA != 0 {
+				fs.fuseConn.AutoInvalData = true
+			} else if out.Flags & linux.FUSE_EXPLICIT_INVAL_DATA != 0 {
+				fs.fuseConn.ExplicitInvalData = true
+			}
+
+			if out.Flags & linux.FUSE_DO_READDIRPLUS != 0 {
+				fs.fuseConn.DoReaddirplus = true
+				if out.Flags & linux.FUSE_READDIRPLUS_AUTO != 0 {
+					fs.fuseConn.ReaddirplusAuto = true
+				}
+			}
+
+			if out.Flags & linux.FUSE_ASYNC_DIO != 0 {
+				fs.fuseConn.AsyncDio = true
+			}
+
+			if out.Flags & linux.FUSE_WRITEBACK_CACHE != 0 {
+				fs.fuseConn.WritebackCache = true
+			}
+
+			if out.Flags & linux.FUSE_PARALLEL_DIROPS != 0 {
+				fs.fuseConn.ParallelDirops = true
+			}
+
+			if out.Flags & linux.FUSE_HANDLE_KILLPRIV != 0 {
+				fs.fuseConn.HandleKillpriv = true
+			}
+
+			// if out.TimeGran > 0 && out.TimeGran <= fuseMaxTimeGranNs {
+				// TODO: figure out how to use this
```

Comment:
It is better to specify a github issue for each TODO. For example, TODO(gvisor.dev/issue/3097)

Reply:
Working on it